### ISSUE TITLE
How to compile tor for orbot？

### ISSUE THREAD

**Issue by LuconYang** [guardianproject/orbot#177]
To add some feature，I rewrote some code of tor. Now I need to configure orbot to reference my customized tor code rather than official tor source code. And I noticed that orbot referenced tor in the build.gradle file as follows:
implementation 'org.torproject:tor-android-binary:0.3.2.10'
How should I compile my tor code and what I need to write in the build.gradle file to reference the compiled tor code?

**Comment by n8fr8** [comment]
May I ask what the features are you are adding? Would they be useful to upstream? Have you asked anyone in the tor community to review them?

You should compile it within this project: https://github.com/n8fr8/tor-android
and then you can use the gradle command line to generate an AAR file (try "uploadArchives" or "release"), which is like a JAR file but with all the native libraries, as well.

From there, you can add this to the OrbotServices build.gradle:

  repositories {
        flatDir {
            dir 'libs'
        }
    }

    implementation(name: 'tor-android-binary', ext: 'aar')

**Comment by LuconYang** [comment]
Thanks for the reply! I add some feature just for some test and that won't change tor's functionality.

I built the project according to the BUILD file in this project: https://github.com/n8fr8/tor-android
and when I run this command:
./tor-droid-make.sh build
I got a error like this:
![image](https://user-images.githubusercontent.com/29349910/47403637-ca045c00-d77c-11e8-9599-aed692d1a7a4.png)
It says it can't find C99 compiler. However, I have installed gcc indeed and it's path is already in the PATH environment varialble. I searched on the internet but can't find a solution.
Could you give some advice for it? Thanks a lot!

**Comment by LuconYang** [comment]
Hi! I solved the privious problem by changing the NDK to a lower version(Revision 15c). After that, the error didn't occur again.

But a new problem arose. I found the change to the Tor source code disappeared after I run the command:  ./tor-droid-make.sh build,   which may mean my changes to the Tor don't become effective.
I guess maybe I need to make some changes to the configuration file. 
Looking forward to your constructive comments!
